kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ReplicaManager.java

```diff
@@ -95,6 +95,8 @@ public CompletableFuture<Map<TopicPartition, ProduceResponse.PartitionResponse>>
                 return;
             }
             if (restTopicPartitionNum == 0) {
+                // If all tasks are sent, cancel the timer tasks to avoid full gc or oom
+                producePurgatory.checkAndComplete(new DelayedOperationKey.TopicPartitionOperationKey(topicPartition));
```

Comment:
I think we should check all partitions in `entriesPerPartition`. Maybe your test scenario only covers the case of 1 partition so it worked well.

Reply:
You can see this line of code: `producePurgatory.tryCompleteElseWatch(delayedProduce, delayedCreateKeys)`, delayedCreateKeys is a many-to-one relationship with delayedProduce, as long as the last partition task is completed, just cancel the delay task once.